### PR TITLE
Create AppSnackbar component and useSnackbar hook

### LINKED ISSUE #121: Implement AppSnackbar component

### PR DESCRIPTION
**Pull request by mhevyk**
https://github.com/idx-academy/spa-orders/assets/109922101/0f0e6602-53a7-46a8-8a8e-0a72b1a23a37

Example of usage:

image

image

image

Unit tests:
image
image

Mutation tests:
image
image

### CODE REVIEW

**Review by D1nviz — APPROVED** [review]

**Review by heshvab — APPROVED** [review]

File: code/src/components/app-snackbar/AppSnackbar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+const mockUseSnackbar = {
+  isOpen: true,
+  config: { message: "Test message", variant: "success" },
+  closeSnackbar: mockCloseSnackbar
+};
+
+describe("AppSnackbar", () => {
+  test("renders correctly", () => {
+    (useSnackbar as jest.Mock).mockReturnValueOnce(mockUseSnackbar);
```
Comment (dmtrth25) [inline code/src/components/app-snackbar/AppSnackbar.test.tsx:21]: we have repeated this logic for tests with `mockReturnValueOnce` and also `renderWithProviders` maybe move it in `beforeEach` block ?

File: code/src/components/app-snackbar/AppSnackbar.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import Snackbar from "@mui/material/Snackbar";
+import Alert from "@mui/material/Alert";
+import useSnackbar from "@/hooks/use-snackbar/useSnackbar";
+
+import "@/components/app-snackbar/AppSnackbar.scss";
+
+export default function AppSnackbar() {
```
Comment (dmtrth25) [inline code/src/components/app-snackbar/AppSnackbar.tsx]: Firstly we define const function and at the bottom use `export default`
Everything should be in one approach like our previous component with export

File: code/src/store/slices/snackbarSlice.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        clearTimeout(state._timerId);
+        delete state._timerId;
+      }
+    },
+    closeSnackbar: (state) => {
+      state.isOpen = false;
+      if (state._timerId) {
+        clearTimeout(state._timerId);
+        delete state._timerId;
+      }
```
Comment (dmtrth25) [inline code/src/store/slices/snackbarSlice.ts:44]: I suggest to create some utility function which used to encapsulate logic for clearing the timer and reducing code duplication. can create above
It's repeated also for `openSnackbar` as you can see

File: code/src/components/app-snackbar/AppSnackbar.test.tsx (hunk 1 above)
Comment (mhevyk) [reply code/src/components/app-snackbar/AppSnackbar.test.tsx:21]: Thanks for useful comment. It was worth refactoring, which I have done.

File: code/src/components/app-snackbar/AppSnackbar.tsx (hunk 2 above)
Comment (mhevyk) [reply code/src/components/app-snackbar/AppSnackbar.tsx]: Definitely. Fixed

File: code/src/store/slices/snackbarSlice.ts (hunk 3 above)
Comment (mhevyk) [reply code/src/store/slices/snackbarSlice.ts:44]: Created utility private function for it

Comment (dmtrth25) [reply code/src/store/slices/snackbarSlice.ts:44]: > Created utility private function for it

I thought about separate `fn` above. But your case also make sense

**Review by dmtrth25 — APPROVED** [review]

**Pull request merged**